shaman/tests/test_util.py

```diff
@@ -105,3 +106,68 @@ def test_pick_last_used_node(self, session, monkeypatch):
         n2.last_used = datetime.datetime.now()
         session.commit()
         assert n1 == util.get_next_node()
+
+
+class TestParseDistroRelease(object):
+
+    @py.test.mark.parametrize('version,codename', [('14.04', 'trusty'), ('16.04', 'xenial'), ('16.10', 'yakkety')])
+    def test_ubuntu_version_gets_both_parsed(self, version, codename):
+        c, v = util.parse_distro_release(version)
+        assert v == version
+        assert c == codename
+
+    @py.test.mark.parametrize('version,codename', [('14.04', 'trusty'), ('16.04', 'xenial'), ('16.10', 'yakkety')])
+    def test_ubuntu_codename_gets_both_parsed(self, version, codename):
+        c, v = util.parse_distro_release(codename)
+        assert v == version
+        assert c == codename
+
+    def test_centos_gets_version_parsed(self):
+        codename, version = util.parse_distro_release('7')
+        assert version == '7'
+        assert codename is None
```

Comment:
This test is now failing because wheezy and centos7 both have the version ``7``. :(